src/value/de.rs

```diff
@@ -420,8 +426,17 @@ impl Value {
         match *self {
             Value::Null => Unexpected::Unit,
             Value::Bool(b) => Unexpected::Bool(b),
-            Value::I64(i) => Unexpected::Signed(i),
-            Value::F64(f) => Unexpected::Float(f),
+            Value::Number(ref n) => {
+                if let Some(u) = n.as_u64() {
+                    Unexpected::Unsigned(u)
+                } else if let Some(i) = n.as_i64() {
+                    Unexpected::Signed(i)
+                } else if let Some(f) = n.as_f64() {
+                    Unexpected::Float(f)
+                } else {
+                    panic!("unexpected number")
```

Comment:
I don't think this should panic. It's either `unreachable!()` or a new Unexpected case.